pkg/vex/sbomref.go

```diff
@@ -0,0 +1,113 @@
+package vex
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/fanal/artifact"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/sbom/core"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+type SBOMReferenceSet struct {
+	Vexes []VEX
+}
+
+func NewSBOMReferenceSet(report *types.Report) (*SBOMReferenceSet, error) {
+
+	if report.ArtifactType != artifact.TypeCycloneDX {
+		return nil, xerrors.Errorf("externalReferences can only be used when scanning CycloneDX SBOMs: %w", report.ArtifactType)
+	}
+
+	var externalRefs = report.BOM.ExternalReferences()
+	urls := parseToURLs(externalRefs)
+
+	v, err := retrieveExternalVEXDocuments(urls, report)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to fetch external VEX document: %w", err)
+	}
+
+	return &SBOMReferenceSet{Vexes: v}, nil
+}
+
+func parseToURLs(refs []core.ExternalReference) []url.URL {
+	var urls []url.URL
+	for _, ref := range refs {
+		if ref.Type == core.ExternalReferenceVEX {
+			val, err := url.Parse(ref.URL)
+			// do not concern ourselves with relative URLs at this point
+			if err != nil || (val.Scheme != "https" && val.Scheme != "http") {
+				continue
+			}
+			urls = append(urls, *val)
+		}
+	}
+	return urls
+}
+
+func retrieveExternalVEXDocuments(refs []url.URL, report *types.Report) ([]VEX, error) {
+
+	logger := log.WithPrefix("vex").With(log.String("type", "externalReference"))
```

Comment:
nit: We basically use the snake case for log parameters.

```suggestion
	logger := log.WithPrefix("vex").With(log.String("type", "external_reference"))
```

Reply:
Done